### PR TITLE
DLQ testing with options feature flag

### PR DESCRIPTION
**Pull request by kneeyo1**
### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/sentry/ingest/consumer/processors.py [hunk 1]
```diff
@@ -169,6 +171,21 @@ def process_event(
     else:
         processing_store = event_processing_store
 
+    if (
+        consumer_type == ConsumerType.Transactions or data.get("type") == "transaction"
+    ) and options.get("transactions.stale_dlq_enabled"):
+        # check timestamp
+        timestamp = message.get("timestamp") if message.get("timestamp") else start_time
```
Comment (kneeyo1) [inline src/sentry/ingest/consumer/processors.py:178]: not sure if message.get('timesstamp") works need to double check

File: src/sentry/ingest/consumer/processors.py [hunk 2]
```diff
@@ -169,6 +171,21 @@ def process_event(
...
+
+        if timestamp and datetime.datetime.now() - timestamp > options.get(
```
Comment (kneeyo1) [inline src/sentry/ingest/consumer/processors.py:180]: these times probably need to be normalized

File: src/sentry/ingest/consumer/processors.py [hunk 3]
```diff
@@ -169,6 +171,21 @@ def process_event(
...
+            "transactions.stale_dlq_threshold_ms"
+        ):
+
+            raise InvalidMessage(
+                message.get("partition"),
+                message.get("offset"),
+                reason="Stale Message",
```
Comment (victoria-yining-huang) [inline src/sentry/ingest/consumer/processors.py:187]: let's not do a space? idk spaces usually scare me in strings

File: src/sentry/ingest/consumer/processors.py (hunk 1 above)
Comment (victoria-yining-huang) [reply src/sentry/ingest/consumer/processors.py:178]: i found in the s[chema repo](https://github.com/getsentry/sentry-kafka-schemas/blob/a31ab4d4537199c72e5ab26689bbd1e8d766e5a5/examples/events/1/errors1.json#L13) that "timestamp" field is on the same level as "type". And in this file the field `type` is being accesed like `data.get("type")`

**Review by victoria-yining-huang — DISMISSED** [review]

File: src/sentry/ingest/consumer/processors.py (hunk 2 above)
Comment (victoria-yining-huang) [reply src/sentry/ingest/consumer/processors.py:180]: what do you mean normalized?

File: src/sentry/ingest/consumer/processors.py (hunk 1 above)
Comment (kneeyo1) [reply src/sentry/ingest/consumer/processors.py:178]: thank you!

**Comment by kneeyo1** [comment]
closing because https://github.com/getsentry/sentry/pull/82322

**Pull request closed without merging**